fedimint-client/src/lib.rs

```diff
@@ -2245,15 +2251,37 @@ impl ClientBuilder {
         Ok(client)
     }
 
-    pub async fn open(self, root_secret: DerivableSecret) -> anyhow::Result<ClientHandle> {
+    pub async fn open(self, pre_root_secret: DerivableSecret) -> anyhow::Result<ClientHandle> {
         let Some(config) = Client::get_config_from_db(&self.db_no_decoders).await else {
             bail!("Client database not initialized")
         };
 
+        if let Some(secret_hash) = self
+            .db_no_decoders()
+            .begin_transaction_nc()
+            .await
+            .get_value(&ClientPreRootSecretHashKey)
+            .await
+        {
+            if pre_root_secret.derive_pre_root_secret_hash() != secret_hash {
+                bail!("Secret hash does not match. Incorrect secret")
+            }
+
+            let mut dbtx = self.db_no_decoders.begin_transaction().await;
+            dbtx.insert_entry(
```

Comment:
If we were to expect any parallelism reading the secret hash and writing it should happen in the same DBTX, but since we have ownership of self here this seems ok?